### PR TITLE
Fix some broken links

### PR DESCRIPTION
**Pull request by eungjun-yi**

### CODE REVIEW

File: docs/kotlin.core/overload-resolution.md [hunk 1]
```diff
@@ -218,7 +218,7 @@ A function is *applicable* for a specific call if and only if the function param
 
 #### Description
 
-Determining function applicability for a specific call is a [type constraint][Type constraints] problem.
+Determining function applicability for a specific call is a [type constraint][Kotlin Type constraints] problem.
```
Comment (ice-phoenix) [inline docs/kotlin.core/overload-resolution.md]: To better follow the capitalization rules, this should be `Kotlin type constraints`

File: docs/kotlin.core/overload-resolution.md [hunk 2]
```diff
@@ -272,7 +272,7 @@ The rest of this section will try to clarify this mechanism in more detail.
 
 When an overload resolution set $S$ is selected and it contains more than one callable, the next step is to choose the most appropriate candidate from these callables.
 
-The selection process uses the [type constraint][Type constraints] system of Kotlin, in a way similar to the process of [determining function applicability][Determining function applicability for a specific call].
+The selection process uses the [type constraint][Kotlin Type constraints] system of Kotlin, in a way similar to the process of [determining function applicability][Determining function applicability for a specific call].
```
Comment (ice-phoenix) [inline docs/kotlin.core/overload-resolution.md]: To better follow the capitalization rules, this should be `Kotlin type constraints`

File: docs/kotlin.core/rtti.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ RTTI affects the semantics of certain expressions, changing their evaluation dep
 
 - [The type checking operator][Type-checking expression]
 - [The cast expression][cast expression], expecially the `as?` operator
-- [Class literals][class literal] and the values they evaluate to
+- [Class literals][Class literal] and the values they evaluate to
```
Comment (ice-phoenix) [inline docs/kotlin.core/rtti.md]: This should link to `Class literals`

**Review by ice-phoenix — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution! If you would be so kind, could you make a few changes to your PR, so that it's even better?

**Comment by eungjun-yi** [comment]
Thanks for your feedback, @ice-phoenix!  I've updated it as you commented.

**Comment by PetukhovVictor** [comment]
@eungjun-yi also please rebase your branch onto `master`. We recently move a lot of commits to `master` from `dev`.

**Comment by eungjun-yi** [comment]
Okay, I've done @PetukhovVictor

**Pull request closed without merging**